### ISSUE TITLE
Adv/Dis extra dice

### ISSUE THREAD

**Issue by Ikaguia** [ElfFriend-DnD/foundryvtt-retroactive-advantage-5e#5]
When toggling between Advantage and Disadvantage, extra dice will be rolled, even tough you already have the necessary 2:
![image](https://user-images.githubusercontent.com/16377028/166956615-a9b0486b-5796-4b07-a499-5b3c9f4219f8.png)
If it helps, here is the active modules list, core and dnd5e version
![image](https://user-images.githubusercontent.com/16377028/166957016-311f8e00-a65f-4671-b441-3711d5a4dc78.png)
Edit: Tested with only this enabled and got the same results

**Comment by krbz999** [comment]
I get this as well. In fact, if I roll initially with adv/disadv, then change to Normal, then change back to adv/disadv, no extra die is added.
Rolling initially as Normal, then changing between adv and disadv adds 1 die each time (changing back to Normal removes all but the first die, as intended).

(dnd5e v1.6.1)

**Comment by akrigline** [comment]
Bizarre. I'll take a look and see if I can reproduce.

**Comment by JacksonBockus** [comment]
I don't know if this is the "correct" solution, but FYI modifying the switch statemnt as follows fixed it for me.

```js
    switch (newAdvMode) {
      case (NORMAL): {
        d20Term.number = 1; //added this line
        d20Term.results = [d20Term.results.shift()]; // keep only the result of the first element of the array
        break;
      }
  
      case (ADVANTAGE): {
        d20Term.modifiers.push('kh');
  
        // if this d20Term doesn't already have more than 1 rolled value, add a new one
        if (d20Term.number === 1) {
          d20Term.number = 2; //added this line
          d20Term.roll();
        }
        break;
      }
  
      case (DISADVANTAGE): {
        d20Term.modifiers.push('kl');
  
        // if this d20Term doesn't already have more than 1 rolled value, add a new one
        if (d20Term.number === 1) {
          d20Term.number = 2; //added this line
          d20Term.roll();
        }
        break;
      }
```

**Comment by akrigline** [comment]
Thanks for that @JacksonBockus, if you want to open a PR that'll help me get this resolved more quickly :)
I haven't yet had a chance to sit down and look into this.

**Comment by akrigline** [comment]
This is bizarre because I swear I've been happily using this for months without this issue.

But now that I try it I reproduce fine.

I wonder if something in core 5e changed with 1.6 or something which messes with some of my assumptions.

Anyway. Released 1.0.1, big thanks to @JacksonBockus for the fix.
